dll/win32/browseui/utility.cpp

```diff
@@ -14,3 +17,57 @@ void operator delete(void *p, UINT_PTR)
 {
     LocalFree(p);
 }
+
+HRESULT SHELL_GetIDListFromObject(IUnknown *punk, PIDLIST_ABSOLUTE *ppidl)
+{
+#if DLL_EXPORT_VERSION >= _WIN32_WINNT_VISTA && 0 // FIXME: SHELL32 not ready yet
+    return SHGetIDListFromObject(punk, ppidl);
+#else
+    HRESULT hr;
+    IPersistFolder2 *pf2;
+    if (SUCCEEDED(hr = punk->QueryInterface(IID_PPV_ARG(IPersistFolder2, &pf2))))
+    {
+        hr = pf2->GetCurFolder(ppidl);
+        pf2->Release();
+    }
+    IPersistIDList *pil;
+    if (FAILED(hr) && SUCCEEDED(hr = punk->QueryInterface(IID_PPV_ARG(IPersistIDList, &pil))))
+    {
+        hr = pil->GetIDList(ppidl);
+        pil->Release();
+    }
+    return hr;
+#endif
+}
+
+static HRESULT SHELL_CompareAbsoluteIDs(LPARAM lParam, PCIDLIST_ABSOLUTE a, PCIDLIST_ABSOLUTE b)
+{
+    IShellFolder *psf;
+    HRESULT hr = SHGetDesktopFolder(&psf);
+    if (FAILED(hr))
+        return hr;
+    hr = psf->CompareIDs(lParam, a, b);
+    psf->Release();
+    return hr;
+}
+
+BOOL SHELL_IsEqualAbsoluteID(PCIDLIST_ABSOLUTE a, PCIDLIST_ABSOLUTE b)
+{
+    return 0 == SHELL_CompareAbsoluteIDs(SHCIDS_CANONICALONLY, a, b);
+}
+
+BOOL SHELL_IsVerb(IContextMenu *pcm, UINT_PTR idCmd, LPCWSTR Verb)
+{
+    HRESULT hr;
+    WCHAR wide[MAX_PATH];
+    if (SUCCEEDED(hr = pcm->GetCommandString(idCmd, GCS_VERBW, NULL, (LPSTR)wide, _countof(wide))))
+        return !lstrcmpiW(wide, Verb);
+
+    char ansi[MAX_PATH], buf[_countof(wide)];
```

Comment:
```suggestion
    CHAR ansi[_countof(wide)], buf[MAX_PATH];
```
The reason why I'm suggesting this change (even though once compiled both buffers will have the same size), is that in your code, `ansi` buffer is related to `wide` (they are the retrieved command string); on the other hand, `buf` is the converted input buffer `Verb`.

Reply:
I understand and I guess I agree. All of it is moot, CDefaultContextMenu caps the verb buffer at 40 or something IIRC (I don't know if that is the correct limit). What I do know is, MAX_PATH is large enough for a registry key name so the verb will never hit this limit.

Reply:
I'm fine with the chosen MAX_PATH in any case, I also recall you already argued in another PR that 3rd-party code that would handle only the ANSI stuff would already cap itself to such a limit.

Reply:
I'm not sure if the other ansi limit was the same thing. Anyway, it does not matter, [Registry element size limits](https://learn.microsoft.com/en-us/windows/win32/sysinfo/registry-element-size-limits) says "Key name: 255 characters" and the verb is a key.

And like I said, CDefaultContextMenu::AddStaticEntriesForKey has a `WCHAR wszName[40];` but I believe that might be a bug. At least there is no reference linked there says a verb is only 40.